readers/elasticsearch/src/test/java/com/criteo/hadoop/garmadon/elasticsearch/cache/ElasticSearchCacheManagerTest.java

```diff
@@ -0,0 +1,222 @@
+    }
+
+    @Test
+    public void do_not_add_container_with_empty_component_in_cache() {
+        EventHeaderProtos.Header header = EventHeaderProtos.Header.newBuilder()
+            .setUsername(username)
+            .setApplicationId(applicationId)
+            .setApplicationName(applicationName)
+            .setAttemptId("attempt_id")
+            .setContainerId(containerId)
+            .setFramework(framework)
+            .setComponent("")
+            .build();
+
+        DataAccessEventProtos.FsEvent fsEvent = DataAccessEventProtos.FsEvent.newBuilder()
+            .setAction(FsAction.WRITE.name())
+            .setDstPath("hdfs://data:8020/var/test/val.lz4")
+            .setUri("hdfs://data:8020")
+            .setHdfsUser("lakeprobes")
+            .setMethodDurationMillis(100L)
+            .build();
+
+        elasticSearchCacheManager.addContainerComponentInCache(new GarmadonMessage(1, 0L, header, fsEvent, null));
+
+        assertNull(elasticSearchCacheManager.cacheContainerComponent.getIfPresent(containerId));
+    }
+
+    @Test
+    public void do_not_add_container_with_unknown_component_in_cache() {
+        EventHeaderProtos.Header header = EventHeaderProtos.Header.newBuilder()
+            .setUsername(username)
+            .setApplicationId(applicationId)
+            .setApplicationName(applicationName)
+            .setAttemptId("attempt_id")
+            .setContainerId(containerId)
+            .setComponent(Component.UNKNOWN.name())
+            .setFramework(framework)
+            .setComponent("")
+            .build();
+
+        DataAccessEventProtos.FsEvent fsEvent = DataAccessEventProtos.FsEvent.newBuilder()
+            .setAction(FsAction.WRITE.name())
+            .setDstPath("hdfs://data:8020/var/test/val.lz4")
+            .setUri("hdfs://data:8020")
+            .setHdfsUser("lakeprobes")
+            .setMethodDurationMillis(100L)
+            .build();
+
+        elasticSearchCacheManager.addContainerComponentInCache(new GarmadonMessage(1, 0L, header, fsEvent, null));
+
+        assertNull(elasticSearchCacheManager.cacheContainerComponent.getIfPresent(containerId));
+    }
+
+    @Test
+    public void enrich_event_with_app_in_the_cache() {
+        elasticSearchCacheManager.addAppEventInCache(msgAppEvent);
+        elasticSearchCacheManager.addContainerComponentInCache(msgFsEvent);
+
+        elasticSearchCacheManager.enrichEvent(eventMap);
+
+        assertEquals(applicationName, eventMap.get("application_name"));
+        assertEquals(framework, eventMap.get("framework"));
+        assertEquals(username, eventMap.get("username"));
+        assertEquals(yarnTags, eventMap.get("yarn_tags"));
+        assertEquals(component, eventMap.get("component"));
+    }
+
+    @Test
+    public void enrich_unknown_component_event_with_app_in_the_cache() {
+        String unknownContainerId = containerId + "_2";
+        eventMap.put("container_id", unknownContainerId);
```

Comment:
The purpose of this container_id field in the map is not obvious. Then reading the rest of the class we king of understand it, but still we need to remember that eventMap is the same fields as the header defined in setUp.
Usually to avoid this and have more readability in tests, I would propose to define header/fsEvents/appEvents factory methods were you can define the fields where a specific value is important and otherwise randomly generated data